torch/fx/passes/dialect/aten/canonicalize.py

```diff
@@ -0,0 +1,43 @@
+from torch.fx.graph_module import GraphModule
+from torch._ops import OpOverload, OpOverloadPacket
+
+def is_canonical(gm: GraphModule) -> bool:
+
+    for node in gm.graph.nodes:
+        if node.op in {"placeholder", "output"}:
+            continue
+
+        if node.op == "call_method":
+            # canonical aten graph couldn't have call_method nodes
+            return False
+
+        if node.op == "get_attr":
+            return False
+
+        if node.op == "call_module":
+            continue
+
+        if node.op == "call_function":
+            func = None
+            if isinstance(node.target, OpOverload):
+                if node.target._overloadname == "out":
```

Comment:
Like Ed said, this technically won't catch everything.

Maybe we should use torchgen for this, e.g. `torchgen.model.NativeFunction.parse(node.target._schema).func.is_out_fn()`

Reply:
nice... this is what I was looking for.